jenkins/helper/aggregate_coverage.py

```diff
@@ -156,23 +218,82 @@ def main():
             JOB_DONE_QUEUE.get()
             count -= 1
         print('jobset finished')
+
+    # terminate workers
     print('sending queue flush command')
     worker_count = max_jobs * 2
     while worker_count > 0:
         JOB_QUEUE.put(('done', 'done', 'done'))
         worker_count -= 1
-
     print('waiting for jobs to exit')
-
     for worker in WORKER_ARRAY:
         print('.')
         worker.join()
-    print('workers collected')
+    print('all workers joined')
     sys.stdout.flush()
     if not last_output.exists():
         print(f'output {str(last_output)} not there?')
+    result_dir = combined_dir / 'result'
+    last_output.rename(result_dir)
+
+    sourcedir = base_dir / 'ArangoDB'
+    # copy the source files from the sourcecode directory
+    for dir_pair in [
+            ['lib'],
+            ['arangosh'],
+            ['client-tools'],
+            ['arangod'],
+            [Path('utils/gdb-pretty-printers'), Path('utils/gdb-pretty-printers')],
+            [Path('enterprise/Enterprise')],
+            [Path('enterprise/tests'), Path('enterprise/tests')]
+    ]:
+        srcdir = sourcedir / dir_pair[0]
+        if srcdir.exists():
+            baselen = len(str(srcdir))
+            dstdir = None
+            if len(dir_pair) == 2:
+                dstdir = coverage_dir / dir_pair[1]
+            else:
+                dstdir = coverage_dir / dir_pair[0]
```

Comment:
It looks like `dir_pair[1]` is not required since it is always equal to `dir_pair[0]`.
So we can set `dstdir` unconditionaly: `dstdir = coverage_dir / dir_pair[0]`.

And also we can iterate through array of string, not array of arrays:
```
for dir in [
            'lib',
            'arangosh',
            ....
```
In this case, it will be just ` dstdir = coverage_dir / dir`.

Reply:
hm, you're right. in the end 0/1 are allways the same.